### PR TITLE
Add IAM Groups resources

### PR DESCRIPTION
**Pull request by Icerzack**
Changes:
* added new resources for IAM: `selectel_iam_group_v1`, `selectel_iam_group_membership_v1`.
* updated other IAM resources according to a new iam-go SDK version.

### CODE REVIEW

**Review by saydamir — CHANGES_REQUESTED** [review]

File: website/docs/r/iam_group_v1.html.markdown [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+
+## Argument Reference
+
+* `name` - (Required) Group name.
+
+* `description` - (Optional) Group description.
+
+* `role` - (Optional) Manages group roles. You can add multiple roles – each role in a separate block. For more information about roles, see the [Roles](#roles) section.
+
+    * `role_name` - (Required) Role name. Available role names are `iam_admin`, `member`, `reader`, `billing`, `object_storage:admin`, and `object_storage_user`.
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply website/docs/r/iam_group_v1.html.markdown:36]: Thanks, removed it

File: website/docs/r/iam_group_v1.html.markdown [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+* Project administrator - `scope` is `project`, `role_name` is `member`.
+
+* Account viewer - `scope` is `account`, `role_name` is `reader`.
+
+* Project viewer - `scope` is `project`, `role_name` is `reader`.
+
+* Object storage admin - `scope` is `project`, `role_name` is `object_storage:admin`.
+
+* Object storage user - `scope` is `project`, `role_name` is `object_storage_user`.
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply website/docs/r/iam_group_v1.html.markdown:60]: Also removed it

File: selectel/resource_selectel_iam_group_membership_v1.go [hunk 3]
```diff
@@ -0,0 +1,222 @@
...
+
+func generateCompositeID(groupID string, userIDs []string) string {
+	sort.Strings(userIDs)
+	concatenated := groupID + ":" + strings.Join(userIDs, ",")
+	encoded := base64.StdEncoding.EncodeToString([]byte(concatenated))
+
+	return encoded
+}
+
+func parseCompositeID(compositeID string) (string, []string, error) {
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_membership_v1.go:169]: Deleted this function and `generateCompositeID`

File: selectel/resource_selectel_iam_group_membership_v1.go [hunk 4]
```diff
@@ -0,0 +1,222 @@
...
+	for _, user := range response.Users {
+		responseUserIDs = append(responseUserIDs, user.KeystoneID)
+	}
+
+	responseServiceUserIDs := make([]string, 0)
+	for _, serviceUser := range response.ServiceUsers {
+		responseServiceUserIDs = append(responseServiceUserIDs, serviceUser.ID)
+	}
+
+	if !containsAll(userIDs, responseUserIDs) || !containsAll(userIDs, responseServiceUserIDs) {
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_membership_v1.go:91]: Ok, deleted it then

File: selectel/resource_selectel_iam_group_membership_v1.go [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+
+	d.SetId("")
+
+	return nil
+}
+
+func generateCompositeID(groupID string, userIDs []string) string {
+	sort.Strings(userIDs)
+	concatenated := groupID + ":" + strings.Join(userIDs, ",")
+	encoded := base64.StdEncoding.EncodeToString([]byte(concatenated))
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_membership_v1.go:164]: Decided to store `group_id` as id

File: selectel/resource_selectel_iam_group_membership_v1.go [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+	if err != nil {
+		return diag.FromErr(errDeletingObject(objectGroupMembership, d.Id(), err))
+	}
+
+	d.SetId("")
+
+	return nil
+}
+
+func diffUsers(oldUsers, newUsers []string) ([]string, []string) {
```
Comment (saydamir) [inline selectel/resource_selectel_iam_group_membership_v1.go:162]: Perhaps we need maps here? These users slices can be quite large.

File: selectel/resource_selectel_iam_group_v1.go [hunk 7]
```diff
@@ -0,0 +1,193 @@
...
+	if diagErr != nil {
+		return diagErr
+	}
+
+	roles, err := convertIAMSetToRoles(d.Get("role").(*schema.Set))
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	log.Print(msgCreate(objectGroup, d.Id()))
```
Comment (saydamir) [inline selectel/resource_selectel_iam_group_v1.go:74]: d.Id() is always empty.

File: selectel/resource_selectel_iam_group_v1.go [hunk 8]
```diff
@@ -0,0 +1,193 @@
...
+		return diagErr
+	}
+
+	log.Print(msgGet(objectGroup, d.Id()))
+	group, err := iamClient.Groups.Get(ctx, d.Id())
+	if err != nil {
+		return diag.FromErr(errGettingObject(objectGroup, d.Id(), err))
+	}
+
+	if group.Group.Roles != nil && len(group.Group.Roles) != 0 {
```
Comment (saydamir) [inline selectel/resource_selectel_iam_group_v1.go:106]: Do we need these checks?

**Review by saydamir — CHANGES_REQUESTED** [review]

File: selectel/resource_selectel_iam_group_membership_v1.go [hunk 9]
```diff
@@ -0,0 +1,197 @@
...
+		if !slices.Contains(newUsers, user) {
+			usersToRemove = append(usersToRemove, user)
+		}
+	}
+
+	return usersToAdd, usersToRemove
+}
+
+// containsAll checks if sliceB is a subset of sliceA.
+func containsAll(sliceA, sliceB []string) bool {
```
[the remark this answers is not in the document]
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_membership_v1.go]: Removed

File: selectel/resource_selectel_iam_group_membership_v1.go (hunk 6 above)
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_membership_v1.go]: Rewrote inner logic to interact with maps and changed incoming params to maps also

File: selectel/resource_selectel_iam_group_v1.go (hunk 7 above)
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_v1.go]: Thanks, added `opts` there to print

File: selectel/resource_selectel_iam_group_v1.go (hunk 8 above)
Comment (Icerzack) [reply selectel/resource_selectel_iam_group_v1.go]: I think no, removed them also

**Review by saydamir — DISMISSED** [review]

**Review by shlykov — APPROVED** [review]

**Pull request merged**